cvat/apps/engine/serializers.py

```diff
@@ -134,66 +195,124 @@ class Meta:
         fields = ('id', 'svg',)
 
 class LabelSerializer(SublabelSerializer):
-    deleted = serializers.BooleanField(required=False, help_text='Delete label if value is true from proper Task/Project object')
+    deleted = serializers.BooleanField(required=False, write_only=True,
+        help_text='Delete the label. Only applicable in the PATCH methods of a project or a task.')
     sublabels = SublabelSerializer(many=True, required=False)
     svg = serializers.CharField(allow_blank=True, required=False)
+    has_parent = serializers.BooleanField(read_only=True, source='has_parent_label', required=False)
 
     class Meta:
         model = models.Label
-        fields = ('id', 'name', 'color', 'attributes', 'deleted', 'type', 'svg', 'sublabels', 'has_parent')
+        fields = (
+            'id', 'name', 'color', 'attributes', 'deleted', 'type', 'svg',
+            'sublabels', 'project_id', 'task_id', 'parent_id', 'has_parent'
+        )
+        read_only_fields = ('id', 'svg', 'project_id', 'task_id')
+        extra_kwargs = {
+            'project_id': { 'required': False, 'allow_null': False },
+            'task_id': { 'required': False, 'allow_null': False },
+        }
 
     def to_representation(self, instance):
         label = super().to_representation(instance)
         if label['type'] == str(models.LabelType.SKELETON):
             label['svg'] = instance.skeleton.svg
+
+        # Clean mutually exclusive fields
+        if not label.get('task_id'):
+            label.pop('task_id', None)
+        if not label.get('project_id'):
+            label.pop('project_id', None)
+
         return label
 
+    def __init__(self, *args, **kwargs):
+        self._local = kwargs.pop('local', False)
+        """
+        Indicates that the operation is called from the dedicated ViewSet
+        and not from the parent entity, i.e. a project or task.
+        """
+
+        super().__init__(*args, **kwargs)
+
     def validate(self, attrs):
+        if self._local and attrs.get('deleted'):
+            raise serializers.ValidationError(
+                'Labels cannot be deleted by updating in this endpoint. '
+                'Please use the DELETE method instead.'
+            )
+
         if attrs.get('deleted') and attrs.get('id') is None:
```

Comment:
@zhiltsov-max, FYI: not related to this PR but..
If I try to remove the label via `PATCH tasks/tid` request with the body
```
{
  "labels": [
    {
      "id": 101,
      "deleted": true
    }
  ]
}
``` 
![image](https://user-images.githubusercontent.com/49038720/219790178-1800d0ad-61b6-4188-958c-c0edfd7dbc5f.png)

Reply:
This seems to require another PR to fix correctly. I've added a workaround for this. Probably, this will be resolved in #5700. We need to remove validate_labels calls in project/task serializers and try to use DB constraints instead. They seem to be broken in the case of labels for some reason.